src/v/kafka/server/handlers/incremental_alter_configs.cc

```diff
@@ -278,10 +280,151 @@ static ss::future<std::vector<resp_resource_t>> alter_topic_configuration(
       });
 }
 
-static ss::future<std::vector<resp_resource_t>>
-alter_broker_configuartion(std::vector<req_resource_t> resources) {
-    return unsupported_broker_configuration<req_resource_t, resp_resource_t>(
-      std::move(resources));
+/**
+ * For configuration properties that might reasonably be set via
+ * the kafka API (i.e. those in common with other kafka implementations),
+ * map their traditional kafka name to the redpanda config property name.
+ */
+inline std::string_view map_config_name(std::string_view input) {
+    return string_switch<std::string_view>(input)
+      .match("log.cleanup.policy", "log_cleanup_policy")
+      .match("log.message.timestamp.type", "log_message_timestamp_type")
+      .match("log.compression.type", "log_compression_type")
+      .default_match(input);
+}
+
+static ss::future<std::vector<resp_resource_t>> alter_broker_configuartion(
+  request_context& ctx, std::vector<req_resource_t> resources) {
+    std::vector<resp_resource_t> responses;
+    responses.reserve(resources.size());
+
+    // If central config is disabled, we cannot set broker properties
+    if (!config::node().enable_central_config()) {
+        std::transform(
+          resources.begin(),
+          resources.end(),
+          std::back_inserter(responses),
+          [](req_resource_t& resource) {
+              return make_error_alter_config_resource_response<resp_resource_t>(
+                resource,
+                error_code::invalid_config,
+                fmt::format(
+                  "changing '{}' broker property isn't currently supported",
+                  resource.resource_name));
+          });
+
+        co_return responses;
+    }
+
+    for (const auto& resource : resources) {
+        cluster::config_update_request req;
+
+        if (!resource.resource_name.empty()) {
+            responses.push_back(
+              make_error_alter_config_resource_response<resp_resource_t>(
+                resource,
+                error_code::invalid_config,
+                "Setting broker properties on named brokers is unsupported"));
+            continue;
+        }
+
+        bool errored = false;
+        for (const auto& c : resource.configs) {
+            auto mapped_name = map_config_name(c.name);
+            auto op = static_cast<config_resource_operation>(
+              c.config_operation);
```

Comment:
I think after cpp17 this is undefined behavior https://wg21.cmeerw.net/cwg/issue1766 unless c.config_operation is already sanitized here and now coming in raw from the wire.

Reply:
Huh, didn't know that (deep down I'll always be a C programmer...) updated with an explicit bounds check on the int before casting it